tee-worker/litentry/core/assertion-build/src/dynamic/contracts/DynamicAssertion.sol

```diff
@@ -113,40 +83,72 @@ abstract contract DynamicAssertion {
         return (value);
     }
 
-    function concatenateStrings(string memory a, string memory b)
-    internal
-    pure
-    returns (string memory)
-    {
-        bytes memory concatenatedBytes = abi.encodePacked(a, b);
-        return string(concatenatedBytes);
+    function GetString(
+        string memory url,
+        string memory jsonPointer,
+        HttpHeader[] memory headers
+    ) internal returns (string memory returnVal) {
+        bytes memory encoded = abi.encode(url, jsonPointer, headers);
+        uint256 encoded_len = encoded.length;
+
+        assembly {
+            if iszero(call(not(0), 0x03EA, 0, add(encoded, 0x20), encoded_len, returnVal, 0x82)) {
+                revert(0, 0)
```

Comment:
Please check how i'm handling string result from `toHex` precompile in this commit:
https://github.com/litentry/litentry-parachain/blob/de51144efb50c910c09dfa63b5a2403a9156ae09/tee-worker/litentry/core/assertion-build/src/dynamic/contracts/DynamicAssertion.sol#L131-L158

We should always advance free memory pointer located under `0x40` address otherwise bytes will be overriden by following memory allocations.

I've also removed stripping first 32bytes of encoded string in toHex precompile see: tee-worker/litentry/core/evm-dynamic-assertions/src/precompiles/to_hex.rs

If that's ok please follow so we have consistent impl.